experimental/ast/options.go

```diff
@@ -23,10 +23,15 @@ import (
 	"github.com/bufbuild/protocompile/internal/arena"
 )
 
-// CompactOptions represents the collection of options attached to a field-like declaration,
+// CompactOptions represents the collection of options attached to a [DeclAny],
 // contained within square brackets.
 //
 // CompactOptions implements [Commas] over its options.
+//
+// # Grammar
+//
+//	CompactOptions := `[` (option `,`)* option? `]`
+//	option         := Path [:=]? Expr?
```

Comment:
Similar to comments above for similar constructs
```suggestion
//	CompactOptions := `[` ( Option `,`? )? `]`
//	Option         := Path ( `:` | `=` )? Expr?
```

I left the interior of `[...]` using `?` instead of `+` since we likely want to gracefully handle empty options.